SpeedyContactAdd/AppDelegate (original).swift

```diff
@@ -0,0 +1,203 @@
+//
+//  AppDelegate.swift
+//  SpeedyContactAdd
+//
+//  Created by Aaron Haughton on 7/16/16.
+//  Copyright © 2016 AlphaGradeINC. All rights reserved.
+//
+
+import UIKit
+import UserNotifications
+import CoreData
+import Contacts
+import WatchConnectivity
+@UIApplicationMain
+
+class AppDelegate: UIResponder, UIApplicationDelegate, WCSessionDelegate {
+    func sessionDidBecomeInactive(_ session: WCSession) {
+        
+    }
+    
+    func sessionDidDeactivate(_ session: WCSession) {
+        
+    }
+    
+    func session(_ session: WCSession, activationDidCompleteWith activationState: WCSessionActivationState, error: Error?) {
+        print("Session is activated")
+    }
+    
+    
+    var session: WCSession!
+    var window: UIWindow?
+    
+    
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey : Any]? = nil) -> Bool {
+        
+        //  PhoneSessionManager.sharedManager.startSession()
+        
+        if WCSession.isSupported() {
+        let session = WCSession.default
+        session.delegate = self
+        session.activate()
+                            }
+        
+        return true
+    }
+    
+    func session(_ session : WCSession, didReceiveMessage message: [String : Any]) {
+        
+        // Receive messages from watch
+            let recipNumberString = message["PhoneNumber"]!
+            let recipNameString = message["RecipientName"]!
+        
+            // Split String into first name and last name
+            
+            let fullName = ("\(recipNameString)")
+            let fullNameArr = fullName.components(separatedBy: " ")
+            
+            let firstName = fullNameArr[0]
+            let lastName = fullNameArr[1]
+        
+            // Creates New Contact
+    
+            let newContact = CNMutableContact()
+            newContact.givenName = ("\(firstName)")
+            newContact.familyName = ("\(lastName)")
+        
+            let number = CNPhoneNumber(stringValue: ("\(recipNumberString)"))
```

Comment:
Can you not assign these directly? `as NSString` would be more efficient than string interpolation here if you can't